contracts/test/integration/IntegrationTestDaiCleanPriceBoundPSM.t.sol

```diff
@@ -0,0 +1,536 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.13;
+
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+import {Vm} from "./../unit/utils/Vm.sol";
+import {DSTest} from "./../unit/utils/DSTest.sol";
+import {CoreV2} from "../../core/CoreV2.sol";
+import {ICoreV2} from "../../core/ICoreV2.sol";
+import {Constants} from "../../Constants.sol";
+import {IVolt, Volt} from "../../volt/Volt.sol";
+import {PCVGuardian} from "../../pcv/PCVGuardian.sol";
+import {MainnetAddresses} from "./fixtures/MainnetAddresses.sol";
+import {OraclePassThrough} from "../../oracle/OraclePassThrough.sol";
+import {PegStabilityModule} from "../../peg/PegStabilityModule.sol";
+
+/// TODO use vm.deal to increase balance in DAI and VOLT,
+/// increase values for fuzzer for more realistic tests
+contract IntegrationTestDaiCleanPriceBoundPSM is DSTest {
+    using SafeCast for *;
+
+    /// reference PSM to test against
+    PegStabilityModule private immutable priceBoundPsm =
+        PegStabilityModule(MainnetAddresses.VOLT_DAI_PSM);
+    PegStabilityModule private cleanPsm;
+
+    ICoreV2 private core = ICoreV2(MainnetAddresses.CORE);
+    IVolt private volt = IVolt(MainnetAddresses.VOLT);
+    IERC20 private dai = IERC20(MainnetAddresses.DAI);
+    IERC20 private underlyingToken = dai;
+
+    uint256 public constant mintAmount = 10_000_000e18;
+    uint256 public constant voltMintAmount = 10_000_000e18;
+
+    OraclePassThrough public oracle =
+        OraclePassThrough(MainnetAddresses.ORACLE_PASS_THROUGH);
+
+    PCVGuardian public pcvGuardian = PCVGuardian(MainnetAddresses.PCV_GUARDIAN);
+
+    Vm public constant vm = Vm(HEVM_ADDRESS);
+
+    uint128 voltFloorPrice = 1.04e18;
+    uint128 voltCeilingPrice = 1.1e18;
+
+    function setUp() public {
+        /// create PSM
+        cleanPsm = new PegStabilityModule(
+            address(core),
+            address(oracle),
+            address(0),
+            0,
+            false,
+            dai,
+            voltFloorPrice,
+            voltCeilingPrice
+        );
+
+        vm.label(address(cleanPsm), "New PSM");
+        vm.label(address(priceBoundPsm), "Existing PSM");
+
+        uint256 balance = dai.balanceOf(
+            MainnetAddresses.DAI_USDC_USDT_CURVE_POOL
+        );
+        vm.prank(MainnetAddresses.DAI_USDC_USDT_CURVE_POOL);
+        dai.transfer(address(this), balance);
+
```

Comment:
minor, but for tests i think we can just try a `vm.deal`, that way we don't need to keep tracking balances, and tests won't fail if balances change in the future

Reply:
I would use that, but it fails anytime I try to use that new functionality